src/mca/pdl/pdlopen/pdl_pdlopen_module.c

```diff
@@ -175,7 +175,26 @@ static int pdlopen_foreachfile(const char *search_path,
 
         dp = opendir(dirs[i]);
         if (NULL == dp) {
-            ret = PMIX_ERR_IN_ERRNO;
+            if (0 == i) {
```

Comment:
I updated that scenario in OMPI so you now get a better error message:
```
$ mpirun -np 2 -mca mca_base_component_path /home/common hostname
--------------------------------------------------------------------------
We were unable to find any usable plugins for the SCHIZO framework. This
framework requires at least one plugin in order to operate. This can be caused
by any of the following:

* we were unable to build any of the plugins due to some combination
  of configure directives and available system support

* no plugin was selected due to some combination of MCA parameter
  directives versus built plugins (i.e., you excluded all the plugins
  that were built and/or could execute)

* the OPAL_PREFIX environment variable is doesn't point to any location that
  includes at least one usable plugin for this framework.

Please check your installation and environment.
--------------------------------------------------------------------------
```